src/object.d

```diff
@@ -3145,19 +3145,23 @@ unittest
 
 /// ditto
 void destroy(T)(ref T obj)
-    if (!is(T == struct) && !is(T == interface) && !is(T == class) && !_isStaticArray!T)
+    if (!isAggregateType!T && !isStaticArray!T && !isPointer!T)
 {
     obj = T.init;
 }
 
-template _isStaticArray(T : U[N], U, size_t N)
+private enum bool isPointer(T) = is(T == U*, U) && !isAggregateType!T;
+
+private enum bool isAggregateType(T) = is(T == struct) || is(T == union) || is(T == class) || is(T == interface);
```

Comment:
Not sure if this is a good idea. private doesn't mean it's not visible, just that you can't use it.

Reply:
I'm sorry, but I don't understand.  Are you saying there will be name collisions even if they can't be used?

Reply:
Can you provide an example that illustrates the problem?

Reply:
It appears that this is no longer a problem. It used to be. So objection withdrawn.

Reply:
It still might confuse, because `isAggregateType` is in phobos. If you forget to include the correct module, you may still see a private symbol error. But this isn't as bad as it was before, where it would interfere with public symbols.